dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/FATSuite.java

```diff
@@ -121,12 +121,12 @@ public class FATSuite {
         if (isWindows && !FATRunner.FAT_TEST_LOCALRUN) {
             // Repeating the full fat for all features may exceed the 3 hour limit on Fyre Windows and causes random build breaks.
             // Skip EE9 on the windows platform when not running locally.
-            // If we are running with a Java version less than 11 have EE8 (EmptyAction) be the lite mode test to run.
+            // If we are running with a Java version less than 11, have EE8 (EmptyAction) be the lite mode test to run.
             repeat = RepeatTests.with(new EmptyAction().conditionalFullFATOnly(EmptyAction.GREATER_THAN_OR_EQUAL_JAVA_11))
                             .andWith(FeatureReplacementAction.EE10_FEATURES().conditionalFullFATOnly(FeatureReplacementAction.GREATER_THAN_OR_EQUAL_JAVA_17))
                             .andWith(FeatureReplacementAction.EE11_FEATURES());
         } else {
-            // If we are running with a Java version less than 11 have EE9 be the lite mode test to run.
+            // If we are running with a Java version less than 11, have EE9 be the lite mode test to run.
             repeat = RepeatTests.with(new EmptyAction().fullFATOnly())
                             .andWith(FeatureReplacementAction.EE9_FEATURES().conditionalFullFATOnly(FeatureReplacementAction.GREATER_THAN_OR_EQUAL_JAVA_11))
```

Comment:
One thing that I don't understand is what EE9's condition is >= 11.  Shouldn't it be Java 8 instead?

Reply:
Same thing for `repeat = RepeatTests.with(new EmptyAction().conditionalFullFATOnly(EmptyAction.GREATER_THAN_OR_EQUAL_JAVA_11))`

Reply:
```
// EE10 requires Java 11.
// EE11 requires Java 17
// If we only specify EE10/EE11 for lite mode it will cause no tests to run with lower Java versions which causes an error.
if (isWindows && !FATRunner.FAT_TEST_LOCALRUN) {
    // Repeating the full fat for all features may exceed the 3 hour limit on Fyre Windows and causes random build breaks.
    // Skip EE9 on the windows platform when not running locally.
    // If we are running with a Java version less than 11, have EE8 (EmptyAction) be the lite mode test to run.
    repeat = RepeatTests.with(new EmptyAction().conditionalFullFATOnly(EmptyAction.GREATER_THAN_OR_EQUAL_JAVA_11))
                    .andWith(FeatureReplacementAction.EE10_FEATURES().conditionalFullFATOnly(FeatureReplacementAction.GREATER_THAN_OR_EQUAL_JAVA_17))
                    .andWith(FeatureReplacementAction.EE11_FEATURES());
    } else {
        // If we are running with a Java version less than 11, have EE9 be the lite mode test to run.
        repeat = RepeatTests.with(new EmptyAction().fullFATOnly())
                    .andWith(FeatureReplacementAction.EE9_FEATURES().conditionalFullFATOnly(FeatureReplacementAction.GREATER_THAN_OR_EQUAL_JAVA_11))
                    .andWith(FeatureReplacementAction.EE10_FEATURES().conditionalFullFATOnly(FeatureReplacementAction.GREATER_THAN_OR_EQUAL_JAVA_17))
                    .andWith(FeatureReplacementAction.EE11_FEATURES());
    }
```

>     The lowest Java we support is Java 8:
>     Windows:
>     Java 8
>         Lite Mode:
>             EmptyAction - Executed!
>             EE10 - Skipped - Requires Java 11
>             EE11 - Skipped - Requires Java 17
> 
>         Full Mode:
>             EmptyAction - Executed!
>             EE10 - Skipped - Requires Java 11
>             EE11 - Skipped - Requires Java 17
> 
>     Java 11
>         Lite Mode:
>             Empty Action - Skipped due to conditionalFullFATOnly
>             EE10 - Executed!
>             EE11 - Skipped - Requires Java 17
>         
>         Full Mode:
>             EmptyAction - Executed!
>             EE10 - Executed!
>             EE11 - Skipped - Requires Java 17
> 
>     Java 17
>         Lite Mode:
>             EmptyAction - Skipped due to conditionalFullFATOnly
>             EE10 - Skipped due to conditionalFullFATOnly
>             EE11 - Executed!
> 
>         Full Mode:
>             EmptyAction - Executed!
>             EE10 - Executed!
>             EE11 - Executed!
> 
> 
>     Not Windows:
>     Java 8
>         Lite Mode:
>             EmptyAction - Skipped because it is fullFATOnly
>             EE9 - Executed
>             EE10 - Skipped - Requires Java 11
>             EE11 - Skipped - Requires Java 17
> 
>         Full Mode:
>             EmptyAction - Executed!
>             EE9 - Executed!
>             EE10 - Skipped - Requires Java 11
>             EE11 - Skipped - Requires Java 17
> 
> 
>     Java 11
>         Lite Mode:
>             EmptyAction - Skipped because it is fullFATOnly
>             EE9 - Skipped due to conditionalFullFATOnly
>             EE10 - Executed!
>             EE11 - Skipped - Requires Java 17
> 
>         Full Mode:
>             EmptyAction - Executed
>             EE9 - Executed!
>             EE10 - Executed!
>             EE11 - Skipped - Requires Java 17
> 
> 
>     Java 17
>         Lite Mode:
>             EmptyAction - Skipped because it is fullFATOnly
>             EE9 - Skipped due to conditionalFullFATOnly
>             EE10 - Skipped due to conditionalFullFATOnly
>             EE11 - Executed!
> 
>         Full Mode:
>             EmptyAction - Executed!
>             EE9 - Executed!
>             EE10 - Executed!
>             EE11 - Executed!